client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -157,12 +163,44 @@ def _predict(
             _utils.raise_for_http_error(response=response)
         return response
 
+    def _single_batch_predict(
+            self,
+            batch: pd.DataFrame,
+            prediction_id: Optional[str] = None,
+    ):
+        """Make prediction, and error propagation."""
+        request_headers = dict()
+        if prediction_id:
+            request_headers.update({'verta-request-id': prediction_id})
+
+        serialized_batch = batch.to_json(orient="records")
+
+        response = self._session.post(
+            self.batch_prediction_url(),
+            headers=request_headers,
+            json=serialized_batch,
+        )
+        if response.status_code in (
+                400,
+                502,
+        ):  # possibly error from the model back end
+            try:
+                data = _utils.body_to_json(response)
+            except ValueError:  # not JSON response; 502 not from model back end
+                pass
+            else:  # from model back end; contains message (maybe)
+                # try to directly print message, otherwise line breaks appear as '\n'
+                msg = data.get("message") or json.dumps(data)
+                raise RuntimeError(f"deployed model encountered an error: {msg}")
+
+        if not response.ok:
+            _utils.raise_for_http_error(response=response)
+        return response
```

Comment:
We may want to consider putting this into a separate function like `response-handler(response: Response) -> Response` or similar and calling it from both places since it is ~16 lines of identical code.  Not a big deal functionally, so I'm not inclined to hold anything up.

Reply:
good idea! Now that I've got the base functionality working, time to clean up(: